Sources/BarcodeScannerController.swift

```diff
@@ -216,6 +215,30 @@ open class BarcodeScannerController: UIViewController {
     super.viewDidAppear(animated)
     animateFocusView()
   }
+  
+  open override func viewWillTransition(to size: CGSize, with coordinator: UIViewControllerTransitionCoordinator) {
+    super.viewWillTransition(to: size, with: coordinator)
+    
+    coordinator.animate(alongsideTransition: { (context) in
+      self.setupFrame()
+    }) { (context) in
+      
+      self.focusView.layer.removeAllAnimations()
+      
+      UIView.animate(withDuration: 0.3, animations: {
+        if self.barCodeFocusViewType == .oneDimension {
+          self.center(subview: self.focusView, inSize: CGSize(width: 280, height: 80))
+        } else {
+          self.center(subview: self.focusView, inSize: CGSize(width: 218, height: 150))
+        }
+      }, completion: { (finished) in
+        if finished {
+          self.animateFocusView()
+        }
+      })
+      
```

Comment:
I reduced the code a lot, I can't notice difference between everything inside a `UIView.animate` and outside. I think it is still good now.